### PR TITLE
Weather app UI

### PR DESCRIPTION
**Pull request by Buhana149**

### CODE REVIEW

File: lib/text_styles/text_style.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+);
+final TextStyle fontSize18 = GoogleFonts.lato(
+  fontSize: 18,
+  color: Colors.white,
+);
+final TextStyle fontSize16 = GoogleFonts.lato(
+  fontSize: 16,
+  color: Colors.white70,
+);
+final BoxDecoration purpleGradient = BoxDecoration(
```
Comment (vlad-buhaescu) [inline lib/text_styles/text_style.dart]: let's move gradients to their own files

File: lib/text_styles/text_style.dart [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+final TextStyle fontSize36 = GoogleFonts.lato(
```
Comment (vlad-buhaescu) [inline lib/text_styles/text_style.dart]: remove numbers from naming, you can use XLarge, Large, Medium, Small, xSmall

File: lib/text_styles/text_style.dart [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  fontSize: 36,
+  color: Colors.white,
+  fontWeight: FontWeight.bold,
+);
```
Comment (vlad-buhaescu) [inline lib/text_styles/text_style.dart:8]: add an empty line after each declaration

**Review by vlad-buhaescu — CHANGES_REQUESTED** [review]

File: lib/pages/forecast_page.dart [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                            ),
+                            SizedBox(width: 15),
+                            Text(
+                              '7 Day Forecast',
+                              style: fontSize30,
+                            )
+                          ],
+                        ),
+                      ),
+                      ListView.builder(
```
Comment (vlad-buhaescu) [inline lib/pages/forecast_page.dart]: this list can be a separate widget with parameters

File: lib/pages/home_page.dart [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+}
+
+class _HomePageState extends State<HomePage> {
+  final WeatherServices _weatherServices = WeatherServices();
+  String _city = 'London';
+  Map<String, dynamic>? _currentWeather;
+
+  @override
+  void initState() {
+    // TODO: implement initState
```
Comment (vlad-buhaescu) [inline lib/pages/home_page.dart]: remove comment

File: lib/pages/home_page.dart [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+                        '${_currentWeather!['current']['temp_c'].round()}°C',
+                    currentCondition:
+                        '${_currentWeather!['current']['condition']['text']}',
+                    maxTemp:
+                        'Max: ${_currentWeather!['forecast']['forecastday'][0]['day']['maxtemp_c'].round()}°C',
+                    minTemp:
+                        'Min: ${_currentWeather!['forecast']['forecastday'][0]['day']['mintemp_c'].round()}°C',
+                  ),
+                  SizedBox(height: 45),
+                  Row(
```
Comment (vlad-buhaescu) [inline lib/pages/home_page.dart]: separate these 2 rows into a new widgets, one for Sunrise - Sunset, one for Humidty - Wind

File: lib/ui_utilities/build_weather_details.dart [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                        child: Container(
+                          padding: EdgeInsets.all(5),
+                          width: 110,
+                          height: 110,
+                          decoration: BoxDecoration(
+                            gradient: LinearGradient(
+                              begin: AlignmentDirectional.topStart,
+                              end: AlignmentDirectional.bottomEnd,
+                              colors: [
+                                Color(0xFF1A2344).withOpacity(0.5),
```
Comment (vlad-buhaescu) [inline lib/ui_utilities/build_weather_details.dart]: separate colors

File: lib/ui_utilities/build_weather_details.dart [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+                               Text(value is String ? value: value.toString(),
+                               style: GoogleFonts.lato(
+                                fontSize: 18,
+                                color: Colors.white,
+                                
+                               ), )
+                            ],
+                          ),
+                        ),
+                        ),),
```
Comment (vlad-buhaescu) [inline lib/ui_utilities/build_weather_details.dart]: reformat file

File: lib/ui_utilities/circular_indicator_ui.dart [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      decoration: BoxDecoration(
+          gradient: LinearGradient(
+              begin: Alignment.topCenter,
+              end: Alignment.bottomCenter,
+              colors: [
+            Color(0xFF1A2344),
```
Comment (vlad-buhaescu) [inline lib/ui_utilities/circular_indicator_ui.dart]: separate all colors

File: lib/ui_utilities/forecast_tile.dart [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+                ),
+                subtitle: Text(
+                  weatherCondition,
+                  style: fontSize16,
+                ),
+                trailing: Text(
+                  maxMinTemp,
+                  style: fontSize18,
+                ),
+              )),
```
Comment (vlad-buhaescu) [inline lib/ui_utilities/forecast_tile.dart]: missing "," and reformat

File: lib/text_styles/text_style.dart (hunk 2 above)
Comment (Buhana149) [reply lib/text_styles/text_style.dart]: done

File: lib/text_styles/text_style.dart (hunk 3 above)
Comment (Buhana149) [reply lib/text_styles/text_style.dart:8]: done

File: lib/text_styles/text_style.dart (hunk 1 above)
Comment (Buhana149) [reply lib/text_styles/text_style.dart]: done

**Review by vlad-buhaescu — CHANGES_REQUESTED** [review]

**Review by vlad-buhaescu — CHANGES_REQUESTED** [review]
let's addres

File: lib/pages/home_page.dart (hunk 5 above)
Comment (Buhana149) [reply lib/pages/home_page.dart]: done

File: lib/pages/home_page.dart (hunk 6 above)
Comment (Buhana149) [reply lib/pages/home_page.dart]: done

File: lib/ui_utilities/build_weather_details.dart (hunk 8 above)
Comment (Buhana149) [reply lib/ui_utilities/build_weather_details.dart]: done

File: lib/ui_utilities/forecast_tile.dart (hunk 10 above)
Comment (Buhana149) [reply lib/ui_utilities/forecast_tile.dart]: done

File: lib/ui_utilities/build_weather_details.dart (hunk 7 above)
Comment (Buhana149) [reply lib/ui_utilities/build_weather_details.dart]: done

File: lib/ui_utilities/circular_indicator_ui.dart (hunk 9 above)
Comment (Buhana149) [reply lib/ui_utilities/circular_indicator_ui.dart]: done

File: lib/pages/forecast_page.dart (hunk 4 above)
Comment (Buhana149) [reply lib/pages/forecast_page.dart]: done

File: lib/pages/forecast_page.dart [hunk 11]
```diff
@@ -70,23 +71,22 @@ class _ForecastPageState extends State<ForecastPage> {
...
-                              iconImage: iconUrl,
-                              avgTemp:
-                                  '${day['date']}\n${day['day']['avgtemp_c'].round()} °C',
-                              weatherCondition: day['day']['condition']['text'],
-                              maxMinTemp:
-                                  'Max:${day['day']['maxtemp_c']} °C\nMin:${day['day']['mintemp_c']}',
-                            );
-                          })
+                      ForecastListviewBuilder(
+                        itemCount: _forecast!.length,
```
Comment (vlad-buhaescu) [inline lib/pages/forecast_page.dart]: avoid force unwrap:
itemCount: _forecast?.length ?? 0,

File: lib/ui_utilities/build_weather_details.dart [hunk 12]
```diff
@@ -19,46 +20,46 @@ class BuildWeatherDetails extends StatelessWidget {
...
+                Icon(
+                  icon,
+                  color: Colors.white,
+                ),
+                SizedBox(
+                  height: 8,
+                ),
+                Text(
+                  label,
+                  style: GoogleFonts.lato(
```
Comment (vlad-buhaescu) [inline lib/ui_utilities/build_weather_details.dart]: let's add this in styles class and use it here as a constant

File: lib/ui_utilities/build_weather_details.dart [hunk 13]
```diff
@@ -19,46 +20,46 @@ class BuildWeatherDetails extends StatelessWidget {
...
+                    color: Colors.white,
+                    fontWeight: FontWeight.bold,
+                  ),
+                ),
+                SizedBox(
+                  height: 8,
+                ),
+                Text(
+                  value is String ? value : value.toString(),
+                  style: GoogleFonts.lato(
```
Comment (vlad-buhaescu) [inline lib/ui_utilities/build_weather_details.dart]: same here

File: lib/ui_utilities/forecast_listview_builder.dart [hunk 14]
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+
+
```
Comment (vlad-buhaescu) [inline lib/ui_utilities/forecast_listview_builder.dart]: extra line

File: lib/ui_utilities/humidity_wind.dart [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+                          icon: Icons.opacity,
+                          value: humidityValue),
+                      BuildWeatherDetails(
+                          label: 'Wind (KPH)',
+                          icon: Icons.wind_power,
+                          value: windValue),
+                    ],
+                  );
+  }
+}
```
Comment (vlad-buhaescu) [inline lib/ui_utilities/humidity_wind.dart]: please reformat file

**Review by vlad-buhaescu — CHANGES_REQUESTED** [review]

File: lib/pages/forecast_page.dart (hunk 11 above)
Comment (Buhana149) [reply lib/pages/forecast_page.dart]: done

File: lib/ui_utilities/build_weather_details.dart (hunk 12 above)
Comment (Buhana149) [reply lib/ui_utilities/build_weather_details.dart]: done

File: lib/ui_utilities/build_weather_details.dart (hunk 13 above)
Comment (Buhana149) [reply lib/ui_utilities/build_weather_details.dart]: done

File: lib/ui_utilities/forecast_listview_builder.dart (hunk 14 above)
Comment (Buhana149) [reply lib/ui_utilities/forecast_listview_builder.dart]: done

File: lib/ui_utilities/humidity_wind.dart (hunk 15 above)
Comment (Buhana149) [reply lib/ui_utilities/humidity_wind.dart]: done

**Review by vlad-buhaescu — APPROVED** [review]

**Pull request merged**